### PR TITLE
Apps: Add egress/dedicated IPs support

### PR DESCRIPTION
**Pull request by danaelhe**
Addresses #1151

API reference:
[https://docs.digitalocean.com/reference/api/api-reference/#operation/apps_create](https://docs.digitalocean.com/reference/api/api-reference/#operation/apps_create
)

Example config:
```
resource "digitalocean_app" "foobar" {
  spec {
    name   = "%s"
    region = "ams"

    static_site {
      name              = "sample-jekyll"
      build_command     = "bundle exec jekyll build -d ./public"
      output_dir        = "/public"
      environment_slug  = "jekyll"
      catchall_document = "404.html"

      git {
        repo_clone_url = "https://github.com/digitalocean/sample-jekyll.git"
        branch         = "main"
      }
    }

    egress {
      type = "DEDICATED_IP"
    }
  }
}
```

Tests passing:
```
x@x terraform-provider-digitalocean % TF_ACC=1 go test -v -run TestAccDigitalOceanApp_Egress ./digitalocean/app
=== RUN   TestAccDigitalOceanApp_Egress
=== PAUSE TestAccDigitalOceanApp_Egress
=== CONT  TestAccDigitalOceanApp_Egress
--- PASS: TestAccDigitalOceanApp_Egress (215.39s)
PASS
ok      github.com/digitalocean/terraform-provider-digitalocean/digitalocean/app        216.154s
```

### CODE REVIEW

File: digitalocean/app/app_spec.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ func appSpecSchema(isResource bool) map[string]*schema.Schema {
 		"features": {
 			Type:        schema.TypeSet,
 			Optional:    true,
+			Computed:    true,
```
Comment (danaelhe) [inline digitalocean/app/app_spec.go:47]: Some apps test non-related to egress were failing. Added some fixes for those.

File: digitalocean/app/resource_app_test.go [hunk 2]
```diff
@@ -246,6 +246,48 @@ func TestAccDigitalOceanApp_StaticSite(t *testing.T) {
...
+					resource.TestCheckResourceAttr(
+						"digitalocean_app.foobar", "spec.0.name", appName),
+					resource.TestCheckResourceAttrSet("digitalocean_app.foobar", "default_ingress"),
+					resource.TestCheckResourceAttrSet("digitalocean_app.foobar", "dedicated_ips.0.ip"),
+					resource.TestCheckResourceAttrSet("digitalocean_app.foobar", "live_url"),
+					resource.TestCheckResourceAttrSet("digitalocean_app.foobar", "active_deployment_id"),
+					resource.TestCheckResourceAttrSet("digitalocean_app.foobar", "updated_at"),
+					resource.TestCheckResourceAttrSet("digitalocean_app.foobar", "created_at"),
+					resource.TestCheckResourceAttr(
+						"digitalocean_app.foobar", "spec.0.static_site.0.catchall_document", "404.html"),
```
Comment (andrewsomething) [inline digitalocean/app/resource_app_test.go:271]: Let's make sure to add a check for the new attribute:

```
resource.TestCheckResourceAttr("digitalocean_app.foobar", "spec.0.egress.0.type", "DEDICATED_IP"),
```

File: digitalocean/app/app_spec.go [hunk 3]
```diff
@@ -57,6 +58,7 @@ func appSpecSchema(isResource bool) map[string]*schema.Schema {
 		"service": {
 			Type:     schema.TypeList,
 			Optional: true,
+			Computed: true,
```
Comment (andrewsomething) [inline digitalocean/app/app_spec.go:61]: I'm not sure we want this at the top level. I'm curious what the issue was that this fixed?

**Review by andrewsomething — COMMENTED** [review]
Looks great! I did have a couple small comments/questions in-line. Besides those, it would be great to get this added to the docs too.

File: digitalocean/app/app_spec.go (hunk 3 above)
Comment (danaelhe) [reply digitalocean/app/app_spec.go]: Hmm....I can't remember and the tests now seem to be fine without this change. I'll leave it out for now, I have a created a Jira ticket to investigate the breaking tests in Apps in general. Will leave that out of scope for this ticket.

**Comment by danaelhe** [comment]
> Looks great! I did have a couple small comments/questions in-line. Besides those, it would be great to get this added to the docs too.

Oh yes, good call on docs 👍

**Review by andrewsomething — APPROVED** [review]
:+1:

**Pull request merged**